lib/burnside/intrinsics.cc

```diff
@@ -276,29 +277,135 @@ mlir::FuncOp MathRuntimeLibrary::getFuncOp(
   return function;
 }
 
+// This helper class computes a "distance" between two function types.
+// The distance represents the argument and result conversions
+// that are required if one use "to" instead of "from".
+// Note that this is not a reflexive distance, and it does not
```

Comment:
I am not sure what "reflexive distance" means.

Doesn't this produce a scoring metric of how many and how complex it is to convert a tuple (of actual arguments) to another tuple (of formal arguments)?

Reply:
> Doesn't this produce a scoring metric of how many and how complex it is to convert a tuple (of actual arguments) to another tuple (of formal arguments)?

Yes, that is a better probably a better wording. Note that the conversion of the result is also considered. Also, the "complexity" is not really what is taken into account here: the "correctness" from a precision point of view is what is taken into account (narrowing vs non narrowing).
 >I am not sure what "reflexive distance" means.

Thanks, I mixed-up my french. "symmetric" is the proper word. It means `d(foo, bar)` and `d(bar, foo)` are not always the same. Which makes sens in this case because if using "y" instead of "x" requires narrowing arguments, it is very likely using "foo" instead of "bar" will not require narrowing the arguments (so in this case `d(bar, foo) < d(foo, bar)` ). Anyway, the point here is to say, the order of arguments  in `FunctionDistance(from, to)` matters.

Reply:
Ok. Makes more sense.